pkg/operator/event_handler.go

```diff
@@ -57,15 +58,24 @@ func (e *EventHandler) OnAdd(obj interface{}, _ bool) {
 }
 
 func (e *EventHandler) OnUpdate(old, cur interface{}) {
-	if old.(metav1.Object).GetResourceVersion() == cur.(metav1.Object).GetResourceVersion() {
+	oldMeta, ok := e.accessor.ObjectMetadata(old)
+	if !ok {
+		return
+	}
+	curMeta, ok := e.accessor.ObjectMetadata(cur)
+	if !ok {
 		return
 	}
 
-	if o, ok := e.accessor.ObjectMetadata(cur); ok {
-		e.logger.Debug(fmt.Sprintf("%s updated", e.objName))
-		e.metrics.TriggerByCounter(e.objName, UpdateEvent)
-		e.enqueueFunc(o.GetNamespace())
+	// Generation value is 0 for ConfigMaps and Secrets.
+	if (curMeta.GetGeneration() == 0 && !hasResourceVersionChanged(oldMeta, curMeta)) ||
+		(curMeta.GetGeneration() != 0 && !hasObjMetaChanged(oldMeta, curMeta)) {
+		return
 	}
```

Comment:
(nit) Would this achieve the same effect? i.e, return if neither the resource version nor object meta has changed. 
```suggestion
	if !hasResourceVersionChanged(oldMeta, curMeta)) && !hasObjMetaChanged(oldMeta, curMeta)) {
		return
	}
```

Reply:
if the generation ==0 check resourceVersion and if the generation != 0 check the metaLabels,, thiis is what we have to do